src/main/java/de/qabel/desktop/storage/cache/CachedBoxNavigation.java

```diff
@@ -36,12 +36,12 @@ public CachedBoxNavigation(T nav, Path path) {
     }
 
     @Override
-    public DirectoryMetadata reloadMetadata() throws QblStorageException {
+    public JdbcDirectoryMetadata reloadMetadata() throws QblStorageException {
         return nav.reloadMetadata();
     }
 
     @Override
-    public void setMetadata(DirectoryMetadata dm) {
+    public void setMetadata(JdbcDirectoryMetadata dm) {
```

Comment:
Not yet unfortunately. The handling of global properties like the temp dir is not there ... we should use more Factories ;)
Currently, the JdbDirectoryMetadata is for example used to extract the location of the database file and the temp dir ... to access the temp dir. We need to fix that first before we can completely use the DirectoryMetadata Interface (the change is caused by a rename + extract interface in the core)